include/oneapi/tbb/blocked_nd_range.h

```diff
@@ -147,6 +147,36 @@ class blocked_nd_range : public blocked_nd_range_impl<Value, N> {
     using base::base;
 };
 
+#if __TBB_CPP17_DEDUCTION_GUIDES_PRESENT
+template <typename Value, unsigned int... Ns,
+          typename = std::enable_if_t<(... && (Ns == 2 || Ns == 3))>>
+blocked_nd_range(const Value (&... dim)[Ns])
+-> blocked_nd_range<Value, sizeof...(Ns)>;
+
+template <typename Value, typename... Values,
+          typename = std::enable_if_t<(... && std::is_same_v<Value, Values>)>>
+blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
+-> blocked_nd_range<Value, 1 + sizeof...(Values)>;
+
+template <typename Value, unsigned int N,
+          typename = std::enable_if_t<(N != 2 && N != 3)>>
+blocked_nd_range(const Value (&)[N])
+-> blocked_nd_range<Value, N>;
+
```

Comment:
What if the argument is explicitly an array of 2 or 3 elements, not a braced init list? Seems there is no deduction guide for it.

Reply:
Passing single array of 2 or 3 elements should be considered as a valid use-case, but having an extra deduction guide for array of 2 or 3 will also cover the braced-init-list of size 2 and 3. 
As we discussed braced-init-list of size 2 and 3 should not be supported because of ambiguity between 1-D range constructed from single blocked_range and 2/3-D range constructed from the array.
This use-case is the main open question in the RFC #1607

Reply:
As discussed in the RFC, added support for both single array and single braced-init-list with preference to multi-dimensional ranges